src/test/java/seedu/address/logic/commands/book/RelatedBookCommandTest.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.logic.commands.book;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPatronAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PATRON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PATRON;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.book.BookRelatedToPatronPredicate;
+import seedu.address.model.patron.Patron;
+import seedu.address.testutil.TypicalLibTask;
+
+/**
+ * Contains integration tests (Interactions with the Model) and unit tests for
+ * {@code RelatedBookCommand}.
+ */
+public class RelatedBookCommandTest {
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(TypicalLibTask.getTypicalLibTask(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_invalidPatronIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPatronList().size() + 1);
+        RelatedBookCommand relatedBookCommand = new RelatedBookCommand(outOfBoundIndex);
+
+        assertCommandFailure(relatedBookCommand, model, Messages.MESSAGE_INVALID_PATRON_DISPLAYED_INDEX);
+    }
+
+    @Test
```

Comment:
Perhaps we could include a test for negative indexes as well so that we can be sure that the command will fail as well for negative indexes? Other than that, I think these tests quite sufficient for the logic of book-related command. 👍

Reply:
Thank you for the advice, will add some negative indexes test cases later on 💯